db/dcrpg/pgblockchain.go

```diff
@@ -814,12 +822,25 @@ func NewChainDBWithCancel(ctx context.Context, dbi *DBInfo, params *chaincfg.Par
 		}
 	}
 
-	// Start the proposal updates handler async method.
-	chainDB.proposalsUpdateHandler()
-
 	return chainDB, nil
 }
 
+// StartPiparserHandler controls how piparser update handler will be initiated.
+// This handler should to be run once only when the first sync after startup completes.
+func (pgb *ChainDB) StartPiparserHandler() {
+	if atomic.LoadUint32(&piParserCounter) != isPiparserRunning {
```

Comment:
Seems like a good use case for a [CAS](https://golang.org/pkg/sync/atomic/#CompareAndSwapUint32).

e.g. 

```go
if atomic.CompareAndSwapUint32(&piParserCounter, 0, 1) {
	// Start the proposal updates handler async method.
	pgb.proposalsUpdateHandler()
	log.Info("Piparser instance to handle updates is now active")
}
```